File: TP1/ScalaComplexV2/src/complex/Conversion.scala
package complex

object Conversion {
  /*
   * Conversions between polar and cartesian coordinates
   */
  def real(radius: Double, azimuth: Angle): Double = radius * Math.cos(azimuth.angle)

  def imag(radius: Double, azimuth: Angle): Double = radius * Math.sin(azimuth.angle)

  def radius(real: Double, imag: Double): Double = Math.sqrt(real * real + imag * imag)

  def azimuth(real: Double, imag: Double): Angle = new Angle(Math.atan2(imag, real))
}

File: TP1/ComplexJava/src/complex/Conversion.java
package complex;

public class Conversion {
	/*
	 * Conversions between polar and cartesian coordinates
	 */
	public static double real(double radius, Angle azimuth){
	    return radius * Math.cos(azimuth.getAngle());
	}
	public static double imag(double radius, Angle azimuth){
	    return radius * Math.sin(azimuth.getAngle());
	}
	public static double radius(double real, double imag){
	    return Math.sqrt(real*real + imag*imag);
	}
	public static Angle azimuth(double real, double imag){
	    return new Angle(Math.atan2(imag, real));  
	}
}
